crates/bevy_gizmos/src/pipeline_2d.rs

```diff
@@ -130,12 +136,103 @@ impl SpecializedRenderPipeline for LineGizmoPipeline {
     }
 }
 
+#[derive(Clone, Resource)]
+struct LineJointGizmoPipeline {
+    mesh_pipeline: Mesh2dPipeline,
+    uniform_layout: BindGroupLayout,
+}
+
+impl FromWorld for LineJointGizmoPipeline {
+    fn from_world(render_world: &mut World) -> Self {
+        LineJointGizmoPipeline {
+            mesh_pipeline: render_world.resource::<Mesh2dPipeline>().clone(),
+            uniform_layout: render_world
+                .resource::<LineGizmoUniformBindgroupLayout>()
+                .layout
+                .clone(),
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, Hash, Clone)]
+struct LineJointGizmoPipelineKey {
+    mesh_key: Mesh2dPipelineKey,
+    joints: GizmoLineJoint,
+}
+
+impl SpecializedRenderPipeline for LineJointGizmoPipeline {
+    type Key = LineJointGizmoPipelineKey;
+
+    fn specialize(&self, key: Self::Key) -> RenderPipelineDescriptor {
+        let format = if key.mesh_key.contains(Mesh2dPipelineKey::HDR) {
+            ViewTarget::TEXTURE_FORMAT_HDR
+        } else {
+            TextureFormat::bevy_default()
+        };
+
+        let shader_defs = vec![
+            #[cfg(feature = "webgl")]
+            "SIXTEEN_BYTE_ALIGNMENT".into(),
+        ];
+
+        let layout = vec![
+            self.mesh_pipeline.view_layout.clone(),
+            self.uniform_layout.clone(),
+        ];
+
+        if key.joints == GizmoLineJoint::None {
+            error!("There is no entry point for line joints with GizmoLineJoints::None. Please consider aborting the drawing process before reaching this stage.");
```

Comment:
I think we can probably have a `error_once` here, if my assumption that this method is called every frame is correct.

Reply:
That function is not called every frame but every time something in the `LineJointGizmoPipelineKey` changes. Basically it is only called the first time you use a certain `GizmoLineJoint`. So changing this to `error_once` shouldn't change anything.

Reply:
Fair, thank you for the explanation :)

Reply:
I'm sorry, but could you explain the error message from the current checks to me 😅? They really don't make sense to me, especially since I didn't change anything about `bevy_log`